### ISSUE TITLE
Blocks with NBT names do not regenerate with their names [1.19]

### ISSUE THREAD

**Issue by BornToDoStuff** [nitnelave/CreeperHeal#74]
Blocks that are able to be named and placed (for instance Shulkers) do not keep their names after regenerating, though the rest of the NBT data seems to be saved successfully.

**Comment by nitnelave** [comment]
Yeah, not surprised here :) It would require a bit of overhaul (though probably not too much?), and updating to the new APIs. I haven't touched this in a long time, and I don't plan to, but if someone wants to come and take over, be my guest.
